Wheels: ADIOS2 -> v2.9.0, blosc2

franzpoeschel opened the pull request:
ADIOS2 v2.9 replaces c-blosc with c-blosc2 as a dependency

TODO:

- [x] Await final release of ADIOS2 v2.9
- [x] Also update HDF5 to 1.14+ for H5Vol layer updates in ADIOS2 v2.9?
  - [x] or disable HDF5 in ADIOS2 for now
- [x] Windows
- [x] Python 3.8+ (3.7 went EOL last month)

franzpoeschel committed:
Update blosc -> blosc2, ADIOS2 -> v2.9.0-rc1

ADIOS2 should be replace by v2.9.0 finally

franzpoeschel committed:
Fix Blosc2 CMake params

franzpoeschel committed:
Guard against CMAKE_OSX_ARCHITECTURES

franzpoeschel committed:
For now, deactivate the pthread patch in ADIOS2

franzpoeschel committed:
Test small patch for blosc2 CMake

franzpoeschel committed:
Add -fPIC flag

franzpoeschel committed:
MacOS compatibility

franzpoeschel committed:
-DADIOS2_USE_Blosc2=ON

franzpoeschel committed:
Build atl and ffs manually to avoid CMake import error

franzpoeschel committed:
Revert "Build atl and ffs manually to avoid CMake import error"

This reverts commit c5469a3fa79715743ff7e177d111a6ec8e1d389a.

franzpoeschel committed:
RC1 -> full release

franzpoeschel committed:
Don't build ADIOS1

franzpoeschel committed:
-DADIOS2_USE_MHS=OFF

franzpoeschel committed:
Use -DADIOS2_INSTALL_GENERATE_CONFIG=OFF workaround

See
https://github.com/ornladios/ADIOS2/issues/3348#issuecomment-1517084004

franzpoeschel committed:
Do not disable find calls

franzpoeschel committed:
Revert "Do not disable find calls"

This reverts commit 7065b8914aeda3aa2ebf40efa882dca49402506b.

franzpoeschel committed:
Try setting ADIOS2_PATH

ax3l committed:
Keep ADIOS1 + c-blosc1

ax3l committed:
Undo `build.yml` Changes

ax3l committed:
ADIOS2: no c-blosc1 pthread issue

ax3l committed:
HDF5: 1.14.1-2

ax3l committed:
macOS CI: Now has 3 cores

ax3l committed:
C-Blosc2 Updates

- CMake options
- Windows
- build stamp

ax3l commented on library_builders.sh:
```diff
@@ -203,6 +211,43 @@ function build_blosc {
+    rm c-blosc2*.tar.gz
+
+    # @todo: proper patch for this
+    sed -E -i.bak 's|if\(\$ENV\{CMAKE_OSX_ARCHITECTURES\} STREQUAL "arm64"\)|if("$ENV{CMAKE_OSX_ARCHITECTURES}" STREQUAL "arm64")|' c-blosc2-*/CMakeLists.txt
```
@franzpoeschel I do not see why that env var needs to be in quotes. Can you elaborate? :)
https://github.com/Blosc/c-blosc2/blob/v2.7.1/CMakeLists.txt#L243

ax3l replied:
Oh, I see:
```
  -- Using ZSTD internal sources.
      "STREQUAL" "arm64"
```

wild!

ax3l replied:
X-ref: https://github.com/Blosc/c-blosc2/pull/525

ax3l commented on library_builders.sh:
```diff
@@ -203,6 +211,43 @@ function build_blosc {
+function build_blosc2 {
+    if [ -e blosc-stamp2 ]; then return; fi
+
+    curl -sLo c-blosc2-v2.7.1.tar.gz \
```
@franzpoeschel Is there a reason you selected 2.7.1 and not the latest 2.9.3 for c-blosc2?

ax3l replied:
Oh, I see:
```
  -- Using ZSTD internal sources.
      "STREQUAL" "arm64"
```

wild!

franzpoeschel replied:
I recall picking the most recent version that I saw. So, no reason probably, no

ax3l committed:
Windows Updates

ax3l committed:
HDF5 Patches

ax3l committed:
C-Blosc 2.9.3 + CMake Patch

ax3l committed:
ARM: H5Detect

ax3l committed:
Windows: Avoid CXX20 in C-Blosc2

ax3l committed:
macOS arm64 HDF5: H5detect cross-compile

ax3l committed:
HDF5 arm64: cross-compiling H5detect

ax3l committed:
[Patch] ADIOS2 c-blosc2 windows.h std::min

ax3l commented:
Windows link error in ADIOS2:
```
2023-07-01T02:44:28.5724342Z LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\adios_reorganize.vcxproj]
2023-07-01T02:44:28.6284514Z libblosc2.lib(blosc2.obj) : error LNK2019: unresolved external symbol compress2 referenced in function blosc_c [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\adios_reorganize.vcxproj]
2023-07-01T02:44:28.6285930Z libblosc2.lib(blosc2.obj) : error LNK2019: unresolved external symbol uncompress referenced in function blosc_d [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\adios_reorganize.vcxproj]
2023-07-01T02:44:28.6575523Z D:\a\openPMD-api\openPMD-api\src\build-adios2\bin\Debug\adios2_reorganize.exe : fatal error LNK1120: 2 unresolved externals [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\adios_reorganize.vcxproj]
2023-07-01T02:44:29.4429936Z   Engine.cpp
2023-07-01T02:44:29.4431298Z LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\bpls.vcxproj]
2023-07-01T02:44:29.5698764Z libblosc2.lib(blosc2.obj) : error LNK2019: unresolved external symbol compress2 referenced in function blosc_c [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\bpls.vcxproj]
2023-07-01T02:44:29.5704186Z libblosc2.lib(blosc2.obj) : error LNK2019: unresolved external symbol uncompress referenced in function blosc_d [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\bpls.vcxproj]
2023-07-01T02:44:29.5725378Z D:\a\openPMD-api\openPMD-api\src\build-adios2\bin\Debug\bpls.exe : fatal error LNK1120: 2 unresolved externals [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\bpls.vcxproj]
```
might be from ZLIB in c-blosc2?
```
...
2023-07-01T02:33:57.9231340Z -- Could NOT find ZLIB_NG (missing: ZLIB_NG_LIBRARY ZLIB_NG_INCLUDE_DIR) 
2023-07-01T02:33:57.9741468Z -- Found ZLIB: C:/Program Files (x86)/zlib/lib/zlibstatic.lib (found version "1.2.13")  
...
```
https://github.com/Blosc/c-blosc2/pull/529

ax3l committed:
[Patch] C-Blosc2: External Zlib

ax3l committed:
[Hack] Windows: Ignore Ext. ZLIB for c-blosc2

Just build again and close eyes.

ax3l commented on library_builders.sh:
```diff
@@ -287,13 +342,12 @@ function build_hdf5 {
 
         curl -sLo osx_cross_configure.patch \
-            https://raw.githubusercontent.com/h5py/h5py/fcaca1d1b81d25c0d83b11d5bdf497469b5980e9/ci/osx_cross_configure.patch
+            https://raw.githubusercontent.com/conda-forge/hdf5-feedstock/ca7b3b40d58cb949638ead75c9527b3fd1112a3f/recipe/patches/osx_cross_configure.patch
```
Ok, the only problem left here is that these two patches for H5detect to use the host native version do not seem to work yet (macOS arm64).

Probably needs to wait until after my vacation.

ax3l replied:
We will stay with HDF5 1.12.2 for now on Unix to move on.

ax3l committed:
ppc64le: Skip C-Blosc & ADIOS1

Due to time limits on Travis-CI.

ax3l committed:
Unix: Keep HDF5 1.12.2 for Now

- issues with cross-compile cannot quickly be solved
  for macOS arm64/aarch64
- disable HDF5 support in ADIOS2 for now

ax3l committed:
[Patch] C-Blosc2 PUBLIC Linkage Zlib

ax3l committed:
Use Blosc2Config.cmake Package

ax3l committed:
Cleanup and `cat` ADIOS2 Config File

ax3l committed:
[Patch] [Hack] CMake --trace-expand

ax3l committed:
Cleanup: cat adios2-config.cmake

ax3l committed:
[Travis] PPC64le: `travis_wait 30`

Takes longer now w/o output to the terminal than the default
wait time of 10min.
https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

ax3l committed:
[ADIOS2] CMake --trace-expand

ax3l committed:
Undo Trace Expand Hacks

ax3l committed:
[Travis] PPC64le: `travis_wait 45`

ax3l committed:
ADIOS2: w/ SST

ax3l committed:
Blosc2: CMake Config PR Merged

ax3l committed:
macOS (x86): 10.15 -> 11.0

The macOS 10.15 build images on GH Actions are gone, so we remove
this EOL platform for good now.

ax3l reviewed (approved):
Whuiii, ok we got it - awesome :)

franzpoeschel commented on library_builders.bat:
```diff
@@ -193,7 +242,8 @@ call :install_buildessentials
 :: build_szip
-call :build_blosc
 call :build_zfp
+call :build_blosc
```
```suggestion
```

I don't see ADIOS1 being built in this file, so Blosc1 should probably be removed, too?

ax3l replied:
Correct, good catch! :)

franzpoeschel commented on library_builders.sh:
```diff
@@ -339,11 +371,16 @@ fi
 
 install_buildessentials
 build_zlib
-build_blosc
```
Here it's probably the opposite. Since this still builds ADIOS1, should be put Blosc1 back in?

ax3l replied:
Discussed in person:
We always skipped ADIOS1 on mac x-compile. Now we skip it also for Linux with ppc64le.

Forgot to remove Blosc1 on Windows (never had ADIOS1 support).

franzpoeschel reviewed (commented):
Great progress, thank you for this, Axel! I think there is still a little inconsistency in where Blosc1 is built by this, otherwise this looks good.

franzpoeschel commented:
The failing Travis jobs look like they're timing out? 
Should we worry about the failing Readthedocs run?

ax3l committed:
Windows: Skip Building Blosc1

Co-authored-by: Franz Pöschel <franz.poeschel@gmail.com>

ax3l commented:
> The failing Travis jobs look like they're timing out?

Looks like sporadic fails, which are unfortunately normal and can be restarted: serverl builds for PPC64le musllinux and manylinux pass.

>  Should we worry about the failing Readthedocs run?

I did not find yet how we can exclude the `wheels` branch from RTD CI builds :sweat:

ax3l commented:
Ah, also time to remove Python 3.7: https://endoflife.date/python

ax3l committed:
Python: 3.8+

Skip older Python releases, 3.7 is EOL since last month.

ax3l committed:
[Patch] ADIOS2: Remove FindBlosc2.cmake

ax3l committed:
C-Blosc 2.10.1

ax3l committed:
Win: ADIOS 2.9.1

ax3l committed:
ADIOS2_Blosc2_PREFER_SHARED=OFF

https://github.com/ornladios/ADIOS2/pull/3715#discussion_r1296413167

ax3l committed:
cibuildwheel 2.12.1 -> 2.14.1

For `pp310-manylinux_aarch64`

ax3l commented:
For the 0.15.2 release, we will need to patch in https://github.com/ornladios/ADIOS2/pull/3769 here
